sw/device/silicon_creator/lib/sigverify/sphincsplus/address.h

```diff
@@ -0,0 +1,208 @@
+ * - Three fields are shared across all formats (layer, tree, and type).
+ *   - The type field indicates which format the address uses.
+ * - All fields are aligned to 32-bit word boundaries within the address
+ *   (meaning some are oversized for the data they contain).
+ *
+ * Common format of addresses:
+ * +---------+---------------+--------------+
+ * | size    | field         | address bits |
+ * +=========+===============+==============+
+ * | 8 bits  | layer address | 0..31        |
+ * +---------+---------------+--------------+
+ * | 64 bits | tree address  | 32..127      |
+ * +---------+---------------+--------------+
+ * | 8 bits  | type          | 128..159     |
+ * +---------+---------------+--------------+
+ * | 96 bits | flexible      | 160..255     |
+ * |         | (see below)   |              |
+ * +---------+---------------+--------------+
+ *
+ * WOTS+ hash address flexible fields:
+ * +---------+------------------+----------+
+ * | 32 bits | key pair address | 160..191 |
+ * +---------+------------------+----------+
+ * | 32 bits | chain address    | 192..223 |
+ * +---------+------------------+----------+
+ * | 32 bits | hash address     | 224..255 |
+ * +---------+------------------+----------+
+ *
+ * WOTS+ public key compression flexible fields:
+ * +---------+-------------------+----------+
+ * | 32 bits | key pair address  | 160..191 |
+ * +---------+-------------------+----------+
+ * | 64 bits | unused (always 0) | 192..255 |
+ * +---------+-------------------+----------+
+ *
+ * Main Merkle tree flexible fields:
+ * +---------+-------------------+----------+
+ * | 32 bits | unused (always 0) | 160..191 |
+ * +---------+-------------------+----------+
+ * | 32 bits | tree height       | 192..223 |
+ * +---------+-------------------+----------+
+ * | 32 bits | tree index        | 224..255 |
+ * +---------+-------------------+----------+
+ *
+ * FORS hash address flexible fields:
+ * +---------+------------------+----------+
+ * | 32 bits | key pair address | 160..191 |
+ * +---------+------------------+----------+
+ * | 32 bits | tree height      | 192..223 |
+ * +---------+------------------+----------+
+ * | 32 bits | tree index       | 224..255 |
+ * +---------+------------------+----------+
+ *
+ * FORS tree root compression flexible fields:
+ * +---------+-------------------+----------+
+ * | 32 bits | key pair address  | 160..191 |
+ * +---------+-------------------+----------+
+ * | 64 bits | unused (always 0) | 192..255 |
+ * +---------+-------------------+----------+
+ */
+
+#include <stdint.h>
+
+#include "sw/device/silicon_creator/lib/sigverify/sphincsplus/params.h"
+
+/**
+ * Hash types for `spx_addr_type_set`.
+ */
+typedef enum spx_addr_type {
+  kSpxAddrTypeWots = 0,
+  kSpxAddrTypeWotsPk = 1,
+  kSpxAddrTypeHashTree = 2,
+  kSpxAddrTypeForsTree = 3,
+  kSpxAddrTypeForsPk = 4,
+  kSpxAddrTypeWotsPrf = 5,
+  kSpxAddrTypeForsPrf = 6,
```

Comment:
In the case of SPHINCS+, these exact values are actually required by the spec (otherwise tests will fail). That's why I added them explicitly even though they're the default. I'll add a comment here explaining that context -- they're really more like required constant values than a true enum.

Reply:
SG, thanks for the clarification. Yes, a comment would be very helpful.